metriq_gym/benchmarks/clops.py

```diff
@@ -0,0 +1,155 @@
+import copy
+from dataclasses import dataclass
+from typing import Optional
+
+import rustworkx as rx
+import numpy as np
+from qbraid import GateModelResultData, QuantumDevice, QuantumJob
+from qiskit import QuantumCircuit
+from qiskit_device_benchmarking.clops.clops_benchmark import append_1q_layer
+
+from metriq_gym.benchmarks.benchmark import Benchmark, BenchmarkData, BenchmarkResult
+from metriq_gym.helpers.execution_time import execution_time
+from metriq_gym.helpers.task_helpers import flatten_job_ids
+from metriq_gym.helpers.topology_helpers import device_topology
+
+
+@dataclass
+class ClopsData(BenchmarkData):
+    pass
+
+
+@dataclass
+class ClopsResult(BenchmarkResult):
+    clops_score: float
+
+
+def create_qubit_map(width, topology_graph: rx.PyGraph, faulty_qubits, total_qubits):
+    """
+    Returns a list of  'width' qubits that are connected based on a coupling map that has
+    bad edges already removed and a list of faulty qubits.  If there is not
+    such a map, raised ValueError
+    """
+    qubit_map = []
+
+    # make coupling map bidirectional so we can find all neighbors
+    cm = copy.deepcopy(topology_graph)
+    for edge in topology_graph.edge_list():
+        cm.add_edge(edge[1], edge[0], None)
+
+    for starting_qubit in range(total_qubits):
+        if starting_qubit not in faulty_qubits:
+            qubit_map = [starting_qubit]
+            new_neighbors = []
+            prospective_neighbors = list(cm.neighbors(starting_qubit))
+            while prospective_neighbors:
+                for pn in prospective_neighbors:
+                    if pn not in qubit_map and pn not in faulty_qubits:
+                        new_neighbors.append(pn)
+                qubit_map = qubit_map + new_neighbors
+                prospective_neighbors = []
+                for nn in new_neighbors:
+                    potential_neighbors = list(cm.neighbors(nn))
+                    for pn in potential_neighbors:
+                        if pn not in prospective_neighbors:
+                            prospective_neighbors.append(pn)
+                new_neighbors = []
+                if len(qubit_map) >= width:
+                    return qubit_map[:width]
+    raise ValueError("Insufficient connected qubits to create set of %d qubits", width)
```

Comment:
[nit]: Use f-string:

```suggestion
    raise ValueError(f"Insufficient connected qubits to create set of {width}| qubits")
```